editor/plugins/item_list_editor_plugin.cpp

```diff
@@ -143,22 +150,46 @@ int ItemListOptionButtonPlugin::get_flags() const {
 	return FLAG_ICON | FLAG_ID | FLAG_ENABLE;
 }
 
-void ItemListOptionButtonPlugin::add_item() {
-	ob->add_item(vformat(TTR("Item %d"), ob->get_item_count()));
+void ItemListOptionButtonPlugin::undoRedo_add_new_item(int item_idx) {
+	ob->add_item(vformat(TTR("Item %d"), item_idx));
+	_change_notify();
+}
+
+void ItemListOptionButtonPlugin::undoRedo_re_add_item(unsigned int position, String name, Ref<Texture2D> icon, bool enabled){
```

Comment:
Why is this `unsigned int`? Everywhere else the position is just `int`

Also, for option buttons the `Id` property exists and is editable - you are not persisting that across undo/redos?